pkg/policy/rule.go

```diff
@@ -179,19 +180,65 @@ func mergeIngressPortProto(ctx *SearchContext, endpoints []api.EndpointSelector,
 	return 1, nil
 }
 
+func traceL3(ctx *SearchContext, peerEndpoints api.EndpointSelectorSlice, direction string) {
+	var result bytes.Buffer
```

Comment:
This code should only be ran if `ctx.Trace != TRACE_DISABLED`, since otherwise it's a no-op for policy computation and we'd be creating these buffers and doing string writing without using the output.

Reply:
Good point. I'm inadvertently achieving this today because only the policy trace cases specify both the source and destination (well, apart from some deniedIdentities logic which @jrajahalme is working on removing right now), but this should be more explicit.